cloudinit/config/schema.py

```diff
@@ -213,20 +241,31 @@ def _schemapath_for_cloudconfig(config, original_content):
             previous_depth = -1
             path_prefix = ''
         if line.startswith('- '):
+            previous_list_idx = '.%d' % (list_index - 1)
```

Comment:
It would be useful to comment in each of these if/else clauses what you're parsing out.

Reply:
done.